core/blockchain.go

```diff
@@ -1145,9 +1142,24 @@ func (self *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 		}
 	}
 
-	if glog.V(logger.Debug) {
+	oldLen := len(oldChain)
+
+	if glog.V(logger.Debug) || oldLen > 63 {
 		commonHash := commonBlock.Hash()
-		glog.Infof("Chain split detected @ %x. Reorganising chain from #%v %x to %x", commonHash[:4], numSplit, oldStart.Hash().Bytes()[:4], newStart.Hash().Bytes()[:4])
+		commonNumber := commonBlock.Number()
+		newLen := len(newChain)
+
+		newLast := newChain[0]
+		newFirst := newChain[newLen-1]
+		oldLast := oldChain[0]
+		oldFirst := oldChain[oldLen-1]
+		glog.Infof("Chain split detected after #%v [%x…]. Reorganising chain (-%v +%v blocks) from #%v-#%v [%x/%x] in favour of #%v-#%v [%x/%x]",
```

Comment:
Style really, but I'd suggest adding the ellipse `…` to the last two brackets too. I.e.

```
glog.Infof("Chain split detected after #%v [%x…]. Reorganising chain (-%v +%v blocks) from #%v-#%v [%…x/%x…] in favour of #%v-#%v [%x…/%x…]",
```